### PR TITLE
jakeware/camera_trigger_fix > master

### PR DESCRIPTION
**Pull request by jakeware**
Waiting on #5.

This little maneuver gets the flea3 camera out of a broken state that it ends up in after boot.  For some reason the trigger configuration is not getting set properly on the camera.  After playing around with the dynamic reconfigure plugin in rqt, it seems like touch any of the trigger params causes it to resend the configuration and everything starts working again.

I know this is super dirty, but I feel like it is better than touching the camera driver.

Resolves: #10

### CODE REVIEW

**Comment by jakeware** [comment]
@wngreene:  This is ready for review.

File: pointgrey/launch/flea3.launch [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<launch>
+   <!-- Determine this using rosrun pointgrey_camera_driver list_cameras.
+       If not specified, defaults to first camera found. -->
+  <arg name="camera_serial" default="0" />
+  <arg name="calibrated" default="0" />
+
+  <!-- Compression can be "jpeg" or "png" -->
+  <arg name="compression" default="png" />
```
Comment (wngreene) [inline pointgrey/launch/flea3.launch]: Does this actually compress the image stream or just set the type of compression used if you subscribe to a compressed topic?

File: pointgrey/launch/flea3.launch [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  <group ns="flea3">
+    <param name="image_raw/compressed/format" value="$(arg compression)" />
+    <param name="image_mono/compressed/format" value="$(arg compression)" />
+    <param name="image_color/compressed/format" value="$(arg compression)" />
+
+    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" />
+
+    <node pkg="nodelet" type="nodelet" name="camera_nodelet"
+          args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
+      <param name="frame_id" value="camera" />
```
Comment (wngreene) [inline pointgrey/launch/flea3.launch]: This should probably be an arg or param.

File: pointgrey/launch/flea3.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+auto_exposure: false
+auto_sharpness: false
+frame_rate: 60
+auto_gain: false
+gain: 4
+auto_shutter: false
+shutter_speed: 0.005
+enable_strobe1: true
+trigger_mode: Mode1_bulb_trigger
+scale: 0.5
```
Comment (wngreene) [inline pointgrey/launch/flea3.yaml]: How does  this file relate to the launch file?

File: ros/src/svis_ros_nodelet.cc [hunk 4]
```diff
@@ -108,12 +111,72 @@ class SVISNodelet : public nodelet::Nodelet {
     camera_buffer_.set_capacity(20);
     camera_strobe_buffer_.set_capacity(10);
 
+    // set pointgrey camera trigger
+    ConfigureCamera();
+
```
Comment (wngreene) [inline ros/src/svis_ros_nodelet.cc]: Is the sole purpose for this call to reset the flea3 state? If so, I would add some comments about why that is necessary.

File: pointgrey/launch/flea3.launch (hunk 2 above)
Comment (jakeware) [inline pointgrey/launch/flea3.launch]: Yeah.  It is probably a dynamic reconfigure parameter, but I don't want to mess with dynamic reconfigure unless necessary.  The easiest way is probably to just make it an arg.

Comment (jakeware) [inline pointgrey/launch/flea3.launch]: Fixed in 84a839169169ff57cbcd9b108378cdae0504292a

File: pointgrey/launch/flea3.yaml (hunk 3 above)
Comment (jakeware) [inline pointgrey/launch/flea3.yaml]: It doesn't actually.  Ted mentioned that loading the params doesn't work.  I should really give it a shot, though.  I have never actually tried it.

File: pointgrey/launch/flea3.launch (hunk 1 above)
Comment (jakeware) [inline pointgrey/launch/flea3.launch]: I actually don't see those compression params getting used.  Maybe image_transport uses them behind the scenes, but I kind of doubt it.

File: pointgrey/launch/flea3.yaml (hunk 3 above)
Comment (wngreene) [inline pointgrey/launch/flea3.yaml]: I would maybe add a comment then saying that it shouldn't be used for anything.

**Pull request merged**